### PR TITLE
improve skip_sync logic

### PR DESCRIPTION
**Pull request by j1rie**

### CODE REVIEW

**Comment by ua0lnj** [comment]
It got worse with early sync+fast sd.
On 25 Hz progressive channels I got frame twitching and bad sync with NVDEC and CPU. VA-API not tested.
I didn't notice any more difference.

**Comment by j1rie** [comment]
"It got worse with early sync+fast sd."
In general or only for NVDEC and CPU?
"On 25 Hz progressive channels"
Which do you use for testing?

**Comment by ua0lnj** [comment]
It seems, for NVDEC and CPU, VAAPI not tested yet.
I have iptv udp channels with 25p, local provider. There are few such channels, but the difference in them is noticeable.
And it doesn’t show up on the recording, live tv only.
I haven't looked at how to fix this yet.
I'll test it again later.

File: softhddev.c [hunk 1]
```diff
@@ -2959,6 +2960,7 @@ int SetPlayMode(int play_mode)
 #ifdef DEBUG
 		    VideoSwitch = GetMsTicks();
 		    Debug(3, "video: new stream start\n");
+		    AudioStarted = 0;
```
Comment (ua0lnj) [inline softhddev.c]: Surely this shouldn't be outside the #ifdef DEBUG ?

Comment (j1rie) [reply softhddev.c]: Yes. My mistake.

**Comment by j1rie** [comment]
Do your problems happen with cuvid, too?
I only test with cuvid, and everything is fine here now (except UHD).

With early audio video buffering is done by skipping audio less (15 * 20 ms), with early sync + fast switch video buffering is poor (but it's fast ;)). Best results with early sync + accurate switch, although not as fast.

UHD is not yet perfect, sometimes glitches after channel switch.

**Pull request merged**